### PR TITLE
🌟 Nova: Webhook Simulator

### PR DESCRIPTION
**Pull request by madmax983**
💡 **The Spark:** Developing and debugging signed webhook endpoints locally is tedious because it traditionally requires using external CLI tools (like the Stripe CLI) or writing custom curl scripts to generate valid HMAC-SHA256 headers manually.

🚀 **The Feature:** Implemented `autumn webhook sim`. This new CLI command accepts a provider preset (`stripe`, `github`, `slack`, `generic`), a target URL, a signing secret, and a JSON payload. It calculates the accurate HMAC signature according to the provider's specific rules (e.g., `Stripe-Signature: t=...,v1=...`, `X-Hub-Signature-256: sha256=...`) and sends the simulated POST request to the local application.

🔭 **The Potential:** This massively improves the Developer Experience by making local iteration on webhook handlers frictionless. Developers can now instantly test edge cases, error conditions, and new event types entirely offline without waiting for third-party deliveries.

⚠️ **Risk:** Low. This is a purely additive CLI utility residing in `autumn-cli/src/webhook.rs` and wired into `autumn-cli/src/main.rs`. It does not modify any core framework logic and leverages the existing `hmac` and `sha2` workspace dependencies. Tests for the provider parsing logic are included and verified.

---
*PR created automatically by Jules for task [9252181054506963862](https://jules.google.com/task/9252181054506963862) started by @madmax983*

### CODE REVIEW

File: autumn-cli/src/webhook.rs [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    match provider {
+        WebhookProvider::Generic => {
+            let mut mac = Hmac::<Sha256>::new_from_slice(secret.as_bytes())
+                .expect("HMAC can take key of any size");
+            mac.update(payload_bytes);
+            let result = mac.finalize();
+            let signature_hex = hex::encode(result.into_bytes());
+
+            req = req.header("X-Webhook-Signature", format!("sha256={signature_hex}"));
+            req = req.header("X-Webhook-Delivery", "sim-delivery-123");
```
[the remark this answers is not in the document]
Comment (madmax983) [reply autumn-cli/src/webhook.rs]: > **P2 Badge Generate unique delivery IDs for each simulated webhook**
> 
> When replay protection is left at its default, Autumn's verifier builds a replay key from the provider, endpoint, and delivery ID and rejects any repeat in `verify_request` for the default 24-hour window. Because the simulator always sends the literal `sim-delivery-123` here (and the GitHub branch uses the same fixed value), the first successful local test poisons the replay cache and subsequent `autumn webhook sim generic`/`github` runs against the same endpoint fail as duplicates instead of letting developers iterate. Generate a fresh ID per invocation or let the caller provide one.
> 
> Useful? React with 👍 / 👎.

@jules
